agcaballero/walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -47,6 +50,42 @@ boolean isReplaceableBy(Availability availability) {
    
   /** the number of minutes in a day */
   private static final int MINUTES_IN_DAY = 24 * 60;
+
+  /**
+   * A private class that holds the necessary elements of availability:
+   * can all mandatory attendees attend and how many optional attendees can't come.
+   */
+  private class Availability {
+    /** holds whether mandatory attendees are available at this time */
+    private boolean areMandatoryAttendeesAllAvailable;
+    /** holds the number of optional attendees that can't attend */
+    private int numberOfOptionalAttendeesUnavailable;
+
+    /**
+     * Constructs the default type of Availability which is that everyone can attend 
+     * hence areMandatoryAttendeesAllAvailable is true and number of optional attendees
+     * that are unavailable is 0.
+     */
+    public Availability() {
+      areMandatoryAttendeesAllAvailable = true;
+      numberOfOptionalAttendeesUnavailable = 0;
+    }
+
+    /**
+     * If any one (or multiple) of the mandatory attendees can't make it, 
+     * sets the mandatory availability of this time to false.
+     */
+    public void mandatoryAttendeeUnavailable() {
```

Comment:
Maybe `setMandatoryAttendeesUnavailable()` so that it's clear this method has a side effect?

Reply:
That makes sense. Thanks!

Reply:
Done